File: mapfilterreduce/mapreducefilter.hs
module Main where 

aList ::  [Integer]
aList = [0, 1, 2]

addOne :: [Integer]
addOne = map (+1) aList

lessThanTwo :: [Integer]
lessThanTwo = filter (<2) aList

reduceToSum :: Integer
reduceToSum = foldl (+) 0 aList
reduceToSumR :: Integer
reduceToSumR = foldr (+) 0 aList
simpleSum ::  Integer
simpleSum = sum aList

main ::  IO ()
main = do
    print aList
    print addOne
    print lessThanTwo
    print reduceToSum
    print reduceToSumR
    print simpleSum
    print aList

File: mapfilterreduce/mapreducefilter.js
'use strict';

// map, reduce and filter are built-in as methods of the Array class in JavaScript
var aList = [0, 1, 2];

var newList = aList.map(function (i) {
  return i + 1;
});
console.log(newList);
console.log(aList);

var reduceToSum = aList.reduce(function (a, b) {
  return a + b;
});
console.log(reduceToSum);
console.log(aList);

var lessThanTwo = aList.filter(function (i) {
  return i < 2;
});
console.log(lessThanTwo);
console.log(aList);

